### PR TITLE
feat: introduce io_v2

### PR DESCRIPTION
**Pull request by ZENOTME**
I will split #242 into smaller PR and this is the first step. 

We will merge the new trait into io_v2 gradually. After all things merge, we can replace io with io_v2. 

This PR:
1. introduce the new write trait in io_v2
2. implement the file writer module

### CODE REVIEW

**Review by liurenjie1024 — COMMENTED** [review]
This looks much clearer.

File: icelake/src/io_v2/mod.rs [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    fn empty() -> Self;
+    fn len(&self) -> usize;
+    fn with_content(&mut self, content: crate::types::DataContentType) -> &mut Self;
+    fn with_equality_ids(&mut self, equality_ids: Vec<i32>) -> &mut Self;
+    fn with_partition(&mut self, partition_value: Option<StructValue>) -> &mut Self;
+    fn extend_res(&mut self, other: Self);
+    fn clear(&mut self) -> Self;
+}
+
+pub trait SingleFileWriter {
```
[the remark this answers is not in the document]
Comment (ZENOTME) [reply icelake/src/io_v2/mod.rs:40]: This trait is also used by IcebergWriter.Such as EqulityDeltaWriter.

File: icelake/src/io_v2/mod.rs [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    fn len(&self) -> usize {
+        self.len()
+    }
+}
+
+// This module provide the test utils for iceberg writer.
+#[cfg(test)]
+// # TODO
+// Remove this allow after the v2 is implemented.
+#[allow(dead_code)]
```
[the remark this answers is not in the document]
Comment (ZENOTME) [reply icelake/src/io_v2/mod.rs:93]: Because the `TestWriter` is used in PartitionWriter. (I will add it later. We can also remove it temporarily.

**Review by liurenjie1024 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**